### PR TITLE
Gsched 627: bug fixes

### PR DESCRIPTION
**Pull request by bryanmiller**
I added various missing class and variable definitions so that it would run without crashing. More tests are needed that the nonsidereal calculations are correct.

### CODE REVIEW

**Comment by bryanmiller** [comment]
Also, we need to merge the new program json files that go with this.

**Comment by bryanmiller** [comment]
Don't test with program GN-2018B-Q-130 at the moment. The target A/2018 W1 is now called C/2018 W1 and is a comet, so the program needs to be edited,

File: scheduler/core/calculations/targetinfo.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class TargetInfo:
     airmass: npt.NDArray[float]
     sky_brightness: npt.NDArray[SkyBackground]
     visibility_slot_idx: npt.NDArray[int]
+    visibility_slot_filter: npt.NDArray[float]
```
Comment (stroncod) [inline scheduler/core/calculations/targetinfo.py:57]: Why the filter needs to be added again? as far as I can tell it was not being used

**Review by stroncod — APPROVED** [review]

File: scheduler/core/calculations/targetinfo.py (hunk 1 above)
Comment (bryanmiller) [reply scheduler/core/calculations/targetinfo.py:57]: I was getting an error in the collector if this wasn't defined.

**Pull request merged**